Geometry/CaloTopology/src/HGCalTopology.cc

```diff
@@ -25,6 +26,11 @@ HGCalTopology::HGCalTopology(const HGCalDDDConstants& hdcons,
     subdet_     = (ForwardSubdetector)(det);
     kHGeomHalf_ = sectors_*layers_;
     types_      = 2;
+  } else if (det < (int)(DetId::HGCalEE)) {
```

Comment:
Why not just check `det==DetId::Forward` here?

Reply:
The value of det is the subdetector type which happens to be DetId::Forward. That is just a coincidence. I am not sure if that could be used.

Reply:
So `det` is not actually a parameter containing the `DetId::det()` field, but only matches up for HGCal?